tracer/src/Datadog.Trace/Util/DbCommandCache.cs

```diff
@@ -39,8 +39,21 @@ internal static class DbCommandCache
 
         public static TagsCacheItem GetTagsFromDbCommand(IDbCommand command)
         {
-            var connectionString = command.Connection?.ConnectionString;
+            IDbConnection dbConnection = null;
+            try
+            {
+                dbConnection = command.Connection;
+            }
+            catch (NotSupportedException nsException)
+            {
+                Log.Warning(nsException, "Connection cannot be retrieved from the command.");
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error trying to retrieve the connection from the command.");
```

Comment:
I'm a bit undecided on the level for this one... In this case, we don't _expect_ to hit it. But if we do, we'll be adding a log for every db span they create. I wonder if we should make this debug too, to avoid flooding the logs?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3693/commits/e43a1c62f6c4c2d1c31d1ef2d6070fedbda02ed3